internal/index/indexoperations/index_test.go

```diff
@@ -64,3 +54,74 @@ func TestListIndexes(t *testing.T) {
 		t.Errorf("output does not match: %s", diff)
 	}
 }
+
+func TestAddIndexSuccess(t *testing.T) {
+	tmpDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	indexName := "foo"
+	localRepo := tmpDir.Path("local/" + indexName)
+	initEmptyGitRepo(t, localRepo, "")
+
+	if err := AddIndex(tmpDir.Path("index"), indexName, localRepo); err != nil {
+		t.Errorf("error adding index: %v", err)
+	}
+	if _, err := os.Stat(tmpDir.Path("index/" + indexName)); err != nil {
+		t.Errorf("error reading newly added index: %s", err)
+	}
+}
+
+func TestAddIndexFailure(t *testing.T) {
+	tmpDir, cleanup := testutil.NewTempDir(t)
+	defer cleanup()
+
+	indexName := "foo"
+	indexRoot := tmpDir.Path("index")
+	if err := AddIndex(indexRoot, indexName, tmpDir.Path("invalid/repo")); err == nil {
+		t.Error("expected error when adding index with invalid URL")
+	}
+
+	localRepo := tmpDir.Path("local/" + indexName)
+	initEmptyGitRepo(t, tmpDir.Path("index/"+indexName), "")
+	initEmptyGitRepo(t, localRepo, "")
+
+	if err := AddIndex(indexRoot, indexName, localRepo); err == nil {
+		t.Error("expected error when adding an index that already exists")
+	}
+
+	if err := AddIndex(indexRoot, "foo/bar", ""); err == nil {
```

Comment:
This invalid index name case was already above. You should co-locate the two cases, so that one has better overview over what's tested.